src/lib/selections.ts

```diff
@@ -512,31 +556,73 @@ function codeToIdSelections(
             : false
         })
         .filter(Boolean)
+
+      /** TODO refactor
+       * selections in our app is a sourceRange plus some metadata
+       * The metadata is just a union type string of different types of artifacts or 3d features 'extrude-wall' 'segment' etc
+       * Because the source range is not enough to figure out what the user selected, so here we're using filtering through all the artifacts
+       * to find something that matches both the source range and the metadata.
+       *
+       * What we should migrate to is just storing what the user selected by what it matched in the artifactGraph it will simply the below a lot.
+       *
+       * In the case of a user moving the cursor them, we will still need to figure out what artifact from the graph matches best, but we will just need sane defaults
+       * and most of the time we can expect the user to be clicking in the 3d scene instead.
+       */
       let bestCandidate
-      entriesWithOverlap.forEach((entry) => {
+      overlappingEntries.forEach((entry) => {
         if (!entry) return
         if (type === 'default' && entry.artifact.type === 'segment') {
           bestCandidate = entry
           return
         }
-        if (
-          type === 'start-cap' &&
-          entry.artifact.type === 'extrudeCap' &&
-          entry?.artifact?.cap === 'start'
-        ) {
-          bestCandidate = entry
+        if (type === 'solid2D' && entry.artifact.type === 'path') {
+          const solid = engineCommandManager.artifactGraph.get(
+            entry.artifact.solid2dId || ''
+          )
+          if (solid?.type !== 'solid2D') return
```

Comment:
In cases where solid2dId is falsey isn't this going to silently fail? Maybe in a follow-up PR more `Error` objects should be returned at various points?

Reply:
if this fails, is basically going to fail to send to the engine what should be highlighted, i.e. you click the solid2d, and you get no visual feedback from the stream that it's been selected, but the FE will still know about the selection so it's not end of the world if it does happen.

In terms of a follow up PR, I'd rather do this from a few lines above this, ya know, this whole section is a bit bad.

```
/** TODO refactor
 * selections in our app is a sourceRange plus some metadata
 * The metadata is just a union type string of different types of artifacts or 3d features 'extrude-wall' 'segment' etc
 * Because the source range is not enough to figure out what the user selected, so here we're using filtering through all the artifacts
 * to find something that matches both the source range and the metadata.
 *
 * What we should migrate to is just storing what the user selected by what it matched in the artifactGraph it will simply the below a lot.
 *
 * In the case of a user moving the cursor them, we will still need to figure out what artifact from the graph matches best, but we will just need sane defaults
 * and most of the time we can expect the user to be clicking in the 3d scene instead.
 */
 
 ```

Reply:
I was thinking just so we can record failures like this / see them in the console and in the future figure out what to do about it but yeah all good